File: rzeka/Assets/rzeka/Runtime/rzeka/interface/serialization/SerializableLooming.cs
using System;
using System.Collections.Generic;

namespace Rzeka
{
  [Serializable]
    public struct SerializableLooming : ISerializableStrandingSpell, ISerlializableBindingSpell
    {
        public Guid guid { get; set; }
        public string title { get; set; }
        public SpellSchool spellSchool { get; set; }
        public string whosName { get; set; }
        public bool wasCast { get; set; }

        public Type conjuredType { get; set; }
        public bool hasMana { get; set; }
        public Dictionary<string, SerializableStranding[]> ingredients { get; set; }
    }
}

File: ui/src/eris/types/common/spells/SerializableLooming.ts
import { SpellSchool } from "./SpellSchoolEnum"
import { Dictionary } from "../Dictionary"
import { Type } from "../Type"
import { Who } from "../Who"
import { ISerializableStrandingSpell } from "./ISerializableStrandingSpell"
import { ISerlializableBindingSpell } from "./ISerlializableBindingSpell"

export class SerializableLooming
    implements ISerializableStrandingSpell, ISerlializableBindingSpell
{
    Who: Who
    conjuredType: Type
    guid: string
    title: string
    spellSchool: SpellSchool
    whosName: string
    wasCast: boolean
    hasMana: boolean
    ingredients: Dictionary<Boolean>
}
